Skau811 favlist custom order frontend

skau811 opened the pull request:
This PR closes #53 by completing the last task of ordering the favourites list in a custom order.

Details of Changes:

**Front-end Adjustments**: Updated the current interface components to be compatible with the updated favourites list data structure.
**Ordering Feature:** Implemented  front-end allowing users to manually order their favourites. Also an attempt has been to update theme customisation feature to ensure the background-image changes according to first movie in the list.  
**Minor Styling:** Incorporated minor CSS enhancements related to the favourites functionality to improve the user experience.

__**Note**_ **for contributors who have already cloned the repo, make sure to run `npm run installAll` after pulling from main, so that dependencies can be easily updated.**_

Testing/Screenshots are as follows:

Favourites list with the sort-by drop-down menu
![image](https://github.com/SE310-1/W.A.K/assets/100451063/edd3f5a7-c942-4162-b55c-53144a637de8)

After reording by top rated:
![image](https://github.com/SE310-1/W.A.K/assets/100451063/ce973f83-1058-4ce0-8409-3a3a3ffa7f4e)

Other minor CSS changes:
Add to favourite button (The use of icons are better way of giving feedback to users instead of only a label text change)
![image](https://github.com/SE310-1/W.A.K/assets/100451063/03335a7c-5c41-4da0-9415-c826ed5b9c98)

Favourites page when the list is empty:
![image](https://github.com/SE310-1/W.A.K/assets/100451063/0042306f-b71a-4dbf-8c06-8e6b2775b485)

skau811 committed:
Refactor existing front-end (favs list components) according to new backend changes

skau811 committed:
Implemented the reordering feature

skau811 committed:
Favourites list now shows the user's rating under each favourite movie

skau811 committed:
Merge branch 'main' into skau811-favlist-custom-order-frontend

skau811 committed:
Add basic style to the drop-down

skau811 committed:
Add basic formatting to grid items in the favourite list

skau811 committed:
Refactor for sonarlint issue

skau811 committed:
Add styling to the grid item subheader

skau811 committed:
Fix sonarlint type declaration issue

skau811 committed:
Refactored existing theme customisation to be compatible with new fav list structure and reordering (first attempt)

skau811 committed:
Fixed theme background image cutomisation when all movies from the favourites list are cleared

skau811 committed:
Styling of some components, adding comments

skau811 committed:
Merge branch 'main' into skau811-favlist-custom-order-frontend

skau811 committed:
Minor css styling changes

skau811 committed:
Minor fix in get fav movies route

skau811 committed:
Minor css styling change of add to fav button

vdav02 reviewed (changes requested):
Hi Sukhleen, thanks for adding these changes in. From my testing, the features match up with what you've described, so we're all good on that end! Unfortunately, I did actually ran into a small issue on my end (seen in the screenshot below) when trying to run the project to test it. 

![image](https://github.com/SE310-1/W.A.K/assets/100258989/ba200283-e6c9-40cf-8fd2-a2e789c2ad4d)

It was an easy fix, as once I ran `npm run installAll`, I no longer encountered the error. Merging this PR shouldn't be an issue for people who are cloning the repo for the first time, as they'll be running `npm run installAll` when they set it up anyway. But for people who have already cloned the repository, I'm thinking that perhaps if you could add a small note in the PR and/or a tag on it (or any other method you deem suitable) to inform people who have already cloned the repo to make sure they execute `npm run installAll` in the terminal, then it should prevent other contributors from running into this issue. Once this is done, I'd definitely be happy to merge it in! Outside of this small import issue, the PR itself is excellent, and adds some awesome functionality and styling to enhance our website - thank you very much!

skau811 commented:
> Hi Sukhleen, thanks for adding these changes in. From my testing, the features match up with what you've described, so we're all good on that end! Unfortunately, I did actually ran into a small issue on my end (seen in the screenshot below) when trying to run the project to test it.
> 
> ![image](https://user-images.githubusercontent.com/100258989/276048146-ba200283-e6c9-40cf-8fd2-a2e789c2ad4d.png)
> 
> It was an easy fix, as once I ran `npm run installAll`, I no longer encountered the error. Merging this PR shouldn't be an issue for people who are cloning the repo for the first time, as they'll be running `npm run installAll` when they set it up anyway. But for people who have already cloned the repository, I'm thinking that perhaps if you could add a small note in the PR and/or a tag on it (or any other method you deem suitable) to inform people who have already cloned the repo to make sure they execute `npm run installAll` in the terminal, then it should prevent other contributors from running into this issue. Once this is done, I'd definitely be happy to merge it in! Outside of this small import issue, the PR itself is excellent, and adds some awesome functionality and styling to enhance our website - thank you very much!

Hi Vishva,
Thanks for your review and highlighting the dependency issue that might happen if existing contributors pull from the main. As requested, I've updated the PR description with a note about running npm run installAll. I have also created a new label for the same and added it to my PR. This should guide other contributors and prevent future issues. 
Thanks again for your feedback, it is highly appreciated.

vdav02 reviewed (approved):
Awesome, thanks so much for making the changes - happy to merge this PR in now!
